src/main/kotlin/com/github/mitchellolsthoorn/testgenie/settings/TestGenieSettingsComponent.kt

```diff
@@ -12,34 +14,167 @@ import javax.swing.JTextField
  */
 class TestGenieSettingsComponent {
     var panel: JPanel? = null
-    var globalTimeoutTextField = JTextField()
-    var showCoverageCheckBox = JCheckBox("Do you want visualised coverage? ")
+    private var showCoverageCheckBox = JCheckBox("Do you want visualised coverage? ")
+    private var sandboxCheckBox = JCheckBox("Execute tests in a sandbox environment")
+    private var assertionsCheckBox = JCheckBox("Create assertions")
+    private var seedTextField = JTextField()
+    //DropDown menu
+    private var algorithmSelector = ComboBox(ContentDigestAlgorithm.values())
+    private var configurationIdTextField = JTextField()
+    private var clientOnThreadCheckBox = JCheckBox("Client on thread")
+    private var junitCheckCheckBox = JCheckBox("Compile and run suite")
+    //There is a limited amount of criteria, but multiple can be selected at once.
+    //Effectively, this requires its own section (or a checkboxed combobox of sorts)
+    private var criterionSeparator = JXTitledSeparator("Criterion selection")
+    private var criterionLineCheckBox = JCheckBox("Line coverage")
+    private var criterionBranchCheckBox = JCheckBox("Branch coverage")
+    private var criterionExceptionCheckBox = JCheckBox("Exception coverage")
+    private var criterionWeakMutationCheckBox = JCheckBox("Mutation coverage")
+    private var criterionOutputCheckBox = JCheckBox("Output coverage")
+    private var criterionMethodCheckBox = JCheckBox("Method coverage")
+    private var criterionMethodNoExceptionCheckBox = JCheckBox("Method no exception coverage")
+    private var criterionCBranchCheckBox = JCheckBox("CBranch coverage")
+    private var minimizeCheckBox = JCheckBox("Minimize test suite after generation")
 
     init {
         panel = FormBuilder.createFormBuilder()
-            .addLabeledComponent(JBLabel("Global timeout "), globalTimeoutTextField, 1, false)
-            .addComponent(showCoverageCheckBox, 1)
+            .addComponent(JXTitledSeparator("General settings"))
+            .addComponent(showCoverageCheckBox, 10)
+            .addComponent(sandboxCheckBox, 10)
+            .addComponent(assertionsCheckBox, 10)
+            .addLabeledComponent(JBLabel("Seed(random if left empty) "), seedTextField, 10, false)
+            .addLabeledComponent(JBLabel("Select search algorithm"), algorithmSelector, 10, false)
+            .addLabeledComponent(JBLabel("Select configuration id (null if left empty) "), configurationIdTextField, 5, false)
```

Comment:
I have a slightly different structure for this part of code (after Lyuben's changes). Is it okay?

Reply:
The change/difference you refer to seems to be mostly stylistic, so all should be good.